contrib/ruby/ext/trilogy-ruby/cast.c

```diff
@@ -142,10 +142,14 @@ rb_trilogy_cast_value(const trilogy_value_t *value, const struct column_info *co
         }
         case TRILOGY_TYPE_DECIMAL:
         case TRILOGY_TYPE_NEWDECIMAL: {
-            // TODO - optimize so we don't have to allocate a ruby string for
-            // decimal columns
-            VALUE str = rb_str_new(value->data, value->data_len);
-            return rb_funcall(rb_mKernel, id_BigDecimal, 1, str);
+            if (column->decimals == 0) {
+                return rb_cstr2inum(value->data, 10);
```

Comment:
@hparker asked me some questions about value->data and now I wonder whether this should be `rb_cstr_parse_inum`, which takes a length.

Reply:
Well not that, since I don't think it's public. But maybe the same thing we do in the else branch?